vendor: bump Pebble to 9af7bb126851

jbowens opened the pull request:
```
9af7bb12 db: close previous WAL before linking next WAL
06b3565a *: Add compactionOutputSplitter, refactor compaction output switch logic
caf883db *: add includesBase argument to ValueMerger.Finish
```

Fixes #54164.
Fixes #54079.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/54194)
<!-- Reviewable:end -->

itsbilal reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 5 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/54194#-:-MGt2w2C0A9bHixC_dYM:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

jbowens commented:
TFTR!

bors r+

jbowens committed:
vendor: bump Pebble to d7a278abeed4

```
d7a278ab compaction: Ensure that user keys aren't split across SSTs when flushing
9af7bb12 db: close previous WAL before linking next WAL
06b3565a *: Add compactionOutputSplitter, refactor compaction output switch logic
caf883db *: add includesBase argument to ValueMerger.Finish
```

Release note: none

jbowens commented:
collided with another PR modifying the vendored repo.

bors r+
